### DISCUSSION TITLE
Add depth to the input channel of YOLO

### DISCUSSION THREAD

**Discussion by PedroPiana** [ultralytics/ultralytics#17273] [Q&A]
Hello, I'm a student, and I'm trying to add a depth channel as an input to YOLO in order to try to improve it. How can I do this?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @PedroPiana, thank you for reaching out to the Ultralytics community 🚀! Your project using Kinect to extract image measurements sounds fascinating. While we can't answer your question directly here, we recommend you look into our [Docs](https://docs.ultralytics.com) for guidance on YOLO customization.

If this is a 🐛 Bug Report, please ensure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so that we can understand the issue better.

For your custom training ❓ question about adding a depth channel, please provide as much detail as possible, including any existing code snippets you've attempted, and check out our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Don't forget to join the vibrant Ultralytics community where you can engage with other users or talk to us directly. Here are some options for you:
- Chat with us live on [Discord](https://ultralytics.com/discord) 🎧.
- Participate in discussions on [Discourse](https://community.ultralytics.com).
- Share insights or find help on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you are using the latest version of the `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). You can do this by running:

```bash
pip install -U ultralytics
```

## Verified Environments

Here are some environments where YOLO can be run:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Follow the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Check the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Use the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all tests are passing for YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/). An Ultralytics engineer will also assist you shortly.

**Comment by glenn-jocher** [comment]
@PedroPiana to add a depth channel to YOLO, you'll need to modify the model architecture to accept additional input channels and adjust the data preprocessing pipeline accordingly. For detailed guidance, you might want to explore the Ultralytics documentation or community forums.
